beacon-chain/sync/rpc.go

```diff
@@ -63,17 +63,21 @@ func (s *Service) registerRPC(baseTopic string, handle rpcHandler) {
 	s.p2p.SetStreamHandler(topic, func(stream network.Stream) {
 		ctx, cancel := context.WithTimeout(s.ctx, ttfbTimeout)
 		defer cancel()
-		defer func() {
-			closeStream(stream, log)
-		}()
+
+		// Resetting after closing is a no-op so defer a reset in case something goes wrong.
+		// It's up to the handler to Close the stream (send an EOF) if
+		// it successfully writes a response. We don't blindly call
+		// Close here because we may have only written a partial
+		// response.
+		defer stream.Reset()
```

Comment:
Please handle this error from stream.Reset(). (Static analysis will not allow the build to complete with unhandled errors)